django/db/migrations/state.py

```diff
@@ -353,6 +353,12 @@ def __init__(self, app_label, name, fields, options=None, bases=None, managers=N
                     'ModelState.fields cannot refer to a model class - "%s.through" does. '
                     'Use a string reference instead.' % name
                 )
+        # Sanity-check that indexes have their names set.
+        for index in self.options['indexes']:
+            try:
+                getattr(index, 'name')
+            except AttributeError:
+                raise ValueError('Indexes passed to ModelState should have their name set.')
```

Comment:
@markush - Does this validation seem good to you about to what we had discussed. The exact behavior we want to check is if `index._name` is set but I don't think using the private variable `_name` over here is a good idea. Any thoughts ?

Also for this validation we don't need to explicitly set the `name` of the index in the `from_model` method like [here](https://github.com/akki/django/pull/1/files#diff-c8a3e1d248f8f5b10af7ac5b1d8479f4R412).